socialdistribution/api/views.py

```diff
@@ -37,8 +44,124 @@ def list(self, request, *args, **kwargs):
+        followers = models.Follower.objects.filter(author = author).all()
+
+        follower_qs = followers.filter(follower_uuid = follower_uuid)
+        if follower_qs.exists():
+            # get author serialize
+            author_qs = Author.objects.filter(uuid = follower_uuid)
+            if author_qs.exists():
+                author = author_qs.get()
+                return serializers.AuthorSerializer(author, context = context).data
+            else:
+                # remote author, GET info
+                remote_node_qs = models.UUIDRemoteCache.objects.filter(uuid = follower_uuid)
+                if remote_node_qs.exists():
+                    # TODO: GET it and set it to response.data
+
+                    node = remote_node_qs.get()
+                    url = (yarl.URL(node.host) / node.prefix / 'author' / follower_uuid / '').human_repr()
+
+                    post_response = requests.get(
+                        url,
+                        headers = { 'Accept': 'application/json' },
+                    )
+
+                    post_response.raise_for_status()
+
+                    if post_response.ok:
+                        return serializers.AuthorSerializer(post_response.json(), context = context).data
+
+        return None
+
+    def retrieve(self, request, *args, **kwargs):
+        """
+        GET {host_url}/author/{author_uuid}/followers/{foreign_uuid}
+
+        Checks if the provided foreign_uuid is a follower of the user
+        """
+        response = Response()
+        # response = super().retrieve(request, *args, **kwargs)
+        foreign_uuid = kwargs['pk']
+
+        follower = self.get_follower(foreign_uuid, { 'request': request })
+
+        if follower is not None:
+            response.data = follower
+        else:
+            return HttpResponseServerError({ 'message': 'Follower exists locally but author cannot be found' })
+
+        return response
+
+    def list(self, request, *args, **kwargs):
+        """
+        GET {host_url}/author/{author_uuid}/followers
+
+        Gets list of authors who are the author_uuid's followers
+        """
+        response = super().list(request, *args, **kwargs)
+        response.data.append({'type': 'followers'})
+        items = list()
+
+        author = Author.objects.filter(uuid = kwargs['author_pk']).get()
+        followers = Follower.objects.filter(author = author).all()
+        for follower in followers:
+            # follower_author = Author.objects.filter(uuid = follower.follower_uuid)
+            follower_uuid = follower.follower_uuid
+            follower_item = self.get_follower(follower_uuid, { 'request': request })
+
+            if follower_item is not None:
+                items.append(follower_item)
+            else:
+                return HttpResponseServerError({ 'message': f'Follower {follower_uuid} exists locally but author cannot be found' })
+
+        response.data.append({'items': items})
+
+        return response
```

Comment:
It seems like the data shape is incorrect. I expect to see:

```json
{
    "type": "followers",
    "items": [
        # ...
    ]
}
```

But what I see is:
```json
[
    {
        "type": "followers"
    },
    {
        "items": [
            # ...
        ]
    }
]
```

This is probably because the parent response data is a list-like object, so you are appending to it when you should be using it as a dictionary.

I encountered this before, so I would suggest to use an `OrderedDict` or `ReturnDict` in a fashion similar to what I did for several of the endpoints:

https://github.com/CMPUT404-F21T0/CMPUT404-Project-BetterSocial/blob/0d4c2f7bb3e3e24e175abdab227c9e97401ffe59/socialdistribution/api/views.py#L29-L44